### DISCUSSION TITLE
Next is suitable for developing large websites??, Especially complex multi interface and data

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#58561] [Help]
### Summary
Next is suitable for developing large websites??, Especially complex multi interface and data ！！
example：
1. The entire page requires SEO,
2. Then, the streaming rendering mentioned in next14 does not meet our requirements because it cannot meet the SEO standards of Chinese browsers
3. 
![image](https://github.com/vercel/next.js/assets/32978177/a34fd12d-5125-4323-987c-04711f04f802)
4. The server can only render HTML for the page after obtaining all the data；We can only handle it this way！
5. If Suspend is used, it will cause problems at the node level and affect SEO
6. 
![img_v3_0257_41a3e1ba-20e8-47f1-8831-78a83463953g](https://github.com/vercel/next.js/assets/32978177/6fe301d2-bcfb-43ef-a5f6-e50d78d9864e)

7. The above figure shows that the node was inserted later. This is related to the node hierarchy and affects the SEO of Chinese browsers

**Comment by parlay96** [comment]
We used next14, but when there are many page interfaces and a large amount of data, our page rendering time is very slow

**Comment by parlay96** [comment]
Does anyone come to discuss it?

**Comment by icyJoseph** [comment]
The docs page you took the screenshots from, also has this section:

- https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#seo

> Since streaming is server-rendered, it does not impact SEO. You can use the [Mobile Friendly Test](https://search.google.com/test/mobile-friendly) tool from Google to see how your page appears to Google's web crawlers and view the serialized HTML

Using a browser, you'll see the streaming happening, you have to access your page like a crawler/bot would.

**Reply by parlay96** [reply]
The problem is that this is the SEO standard on Google, not a Chinese browser

**Reply by icyJoseph** [reply]
- What do you mean with SEO.
- What is the SEO standard you are referring to.

Streaming is an HTTP2 thing, to be able to send chunks of content, rather one big payload. The framework is supposed to understand that if the request comes from a bot/crawler, who just wants to get the page's contents to index them or whatever, then it can just send one big chunk to it. 

- https://github.com/vercel/next.js/discussions/43505

What is confusing me a bit here is, let me phrase this very dry, why does Search Engine Optimization, impact the actual browsing session experience? The SEO is made by bots/crawlers etc, what matters is what they see, not necessarily, what your browser is showing to you. Some sites even detect the bots/crawlers, and give them different content, than the content they share with "humans". 

So could you expand in detail, what is the problem you are facing? How is it practically affecting your business? What is really worrying you?

**Reply by parlay96** [reply]
Can you take a look at my reply below? I am very confused now, can I solve my problem next!

**Comment by parlay96** [comment]
@icyJoseph Firstly, our webpage is very complex with many request interfaces, and our webpage speed is very slow. I want to solve this problem through streaming rendering. But I see that the node hierarchy of streaming rendering is disordered. I just want to solve the problem of fast rendering UI for large web pages. At the same time, our entire page requires SSR，   Thank you for your answer！

**Comment by parlay96** [comment]
@icyJoseph 1. Then, the streaming rendering mentioned in next14 does not meet our requirements because it cannot meet the SEO standards of Chinese browsers
2. 
(https://github.com/vercel/next.js/assets/32978177/8a4cdee6-17ae-4946-ab0f-e836592a45f7)
3. The above figure shows that the node was later inserted. This is related to the node hierarchy and affects the SEO of China browsers

**Reply by icyJoseph** [reply]
I am struggling to understand "SEO of X browsers". SEO is done by Bots and Crawling engines. 

Could you link me to some resource explaining the issue? Maybe it is something I haven't learned yet. 

> The above figure shows that the node was later inserted. This is related to the node hierarchy and affects the SEO of China browsers

This is happening in a browser, not from a bot's perspective. I'll run some tests later today to see the bot's perspective in Next 14.
